src/interface/interface.cpp

```diff
@@ -1359,10 +1356,7 @@ int saveConfig(char* filename)
 	}
 
 	fclose(fp);
-	if ( mallocd )
-	{
-		free(filename);
-	}
+	free(filename);
```

Comment:
Missing a couple other frees, such as on line 1098.

Reply:
To be fair, that memory leak was there before me. I didn't do a full audit, the code has more issues than I really want to dive into today.

I did fix the const constraints, though, so that's progress.

Reply:
Fair enough, that's true; can we touch up that lose end while we're here? ;)

Reply:
Already done. 👉😉👉